Open Pull Request

MsFairyViolet opened the pull request:
…start point

MsFairyViolet committed:
Initial setup: alter standard files included in create-react-app for start point

MsFairyViolet committed:
Compile data: a spreadsheet with title, genre, year, director, imdb rating and a short description, to be used as a mock file and in a database later.

MsFairyViolet committed:
Create mock JSON data file: to work with actual data before there is a database

MsFairyViolet committed:
Remove invalid path from index

MsFairyViolet committed:
Re-add valid files

MsFairyViolet committed:
Create a basic setup: file structure, basic rendering of items, basic styling

Closes: #3

MsFairyViolet committed:
Fetch data from mock.json and display necessary information in individual moviecards

MsFairyViolet committed:
Created different movie card render based on watched status

TODO: improve movie card visuals

MsFairyViolet committed:
xModification to Excel database

TODO: rerender json file before final commit

MsFairyViolet committed:
WIP: Improve Movie Card Visuals

MsFairyViolet committed:
Improve Movie Card Visuals: basic setup

MsFairyViolet committed:
Setup for sub movie functionality

MsFairyViolet committed:
Implement Search Functionality

MsFairyViolet committed:
Feature: Movie Info button and overlay

MsFairyViolet committed:
Abilty to split all movies into standalone and series

MsFairyViolet committed:
Start on Submovies functionality

TODO:
- Generic SeriesCard Title
- All movies watch in series trigger graphical change
- Submovies numbered
- Submovies displayed under SeriesCard
- SeriesCards mixed with MovieCards in MovieList

rmsp committed:
Started backend

MsFairyViolet committed:
Redesign SubMovies Functionality: 1 array with objects, not 2 seperate arrays

MsFairyViolet committed:
Unwatched Series functionality: if 1 movie is unwatched, the whole series is unwatched

MsFairyViolet committed:
Render First Unwatched Series Information as Title on SeriesCard

MsFairyViolet committed:
Implement SubMovieCard

- Displays number based on the series index
- Styles based on watched
- General styling

TODO:
- Movie Info Buttons
- Styling the gaps in the list

MsFairyViolet committed:
Style: Similar behaviour submovie cards on Desktop and Mobile

MsFairyViolet committed:
Implement: submovie functionality & series info card

TODO:
- generic series title
- improved styling (info card and movie list)

MsFairyViolet committed:
Fix Info Button on Series and Movies Card
Redesign code

MsFairyViolet committed:
Merge MovieInfoCard and SeriesInfoCard into one component

MsFairyViolet committed:
Style cards

MsFairyViolet committed:
Merge different Movie/Series/Watched/Unwatched Components into 1 component

MsFairyViolet committed:
Change to Spreadsheet

MsFairyViolet committed:
Rescale title if it is too long

MsFairyViolet committed:
Fix Info Buttons
Style Tabs on InfoCard

MsFairyViolet committed:
Search Functionality Changed to include genres and display series correctly

MsFairyViolet committed:
Style: dropshadows added

MsFairyViolet committed:
Edit Spreadsheet

MsFairyViolet committed:
Implement SubMovie fold-out functionality
Improve styling fold-out and close buttons

MsFairyViolet committed:
Small Style edits

MsFairyViolet committed:
Start implementing Color Palette for Watched Status

MsFairyViolet committed:
Implement Colored Cards: random color when watched for cards, submovies and tabs in the info card
Additional styling

MsFairyViolet committed:
Search functionality for movies part of series: info card now shows the rest of the series in tabs

MsFairyViolet committed:
Add Clear Search Button
Additional Styling

MsFairyViolet committed:
Fetch Data from API

MsFairyViolet committed:
Handle Loading and Errors from API calls, styled

MsFairyViolet committed:
Save Commit
After API fetch
Before Click-To-Watch Functionality

MsFairyViolet committed:
WIP Click-To-Watch

ISSUE: Clicking the sub/info button also triggers the whole area, shouldn't!

rmsp committed:
Fix n+1 query in for findAll movies

rmsp committed:
Made gradle wrapper executable

rmsp committed:
Ensure UTF-8 encoding in spring-boot

MsFairyViolet committed:
Replace mock data.json by API fetch
Click-to-watch patches watched-status and color to database
Prevent re-patch
Rewrite color functionality to render from database, not from state
Color is the same across the whole movie/series, across all elements

MsFairyViolet committed:
Fix issue where subcard could not mark movie as watched

MsFairyViolet committed:
Fix Info and Sub buttons interfering with click-to-watch

MsFairyViolet committed:
Implement Confirm Action and Window

MsFairyViolet committed:
Change to spreadsheet: Harry Potter movie order was wrong

MsFairyViolet committed:
Remove unnecessary files

MsFairyViolet committed:
Changes to Spreadsheet
Render new final CSV

MsFairyViolet committed:
Render new CSV

MsFairyViolet committed:
New CSV, sorted alfabetically on title

MsFairyViolet committed:
Long title on web made smaller so it fits

MsFairyViolet committed:
Spreadsheet and CSV Change
Remove errors out of the series id
Remove a duplicate movie

MsFairyViolet committed:
Fix special characters

Seed data is now always parsed as UTF8

MsFairyViolet committed:
Spreadsheet changes
Reloaded database

MsFairyViolet committed:
Style polish
Responsiveness tested and small changes made
Threshold long titles is lower
Info card doesn't show double minutes and ranking is a decimal number

MsFairyViolet committed:
Style fix: Long titles are even smaller on desktop

MsFairyViolet committed:
Clean up code, empty white spaces and unneccessary comments removed

rmsp reviewed (changes requested):
It looks really good, I like the simple setup of the components. It's really clear what goes where. The integration with the backend works really cleanly, and the state you defined for it really lends itself to clean structure without too much passing things around.

What I like a little bit less is the "is this a series or a single movie" that is sprinkled across everything. We did a hard commit to this data structure: "let's figure it out in the front end". I think that is the exact result of this decision, and it's fine for now. If we were to do this again with a more complicated data structure we'd probably have to make some other decisions.

Good job :rocket:

MsFairyViolet committed:
Pull request changes:
- Remove a tester file
- Remove the mock data file
- Remove a few unnecessary variable declarations
- Rewrite a check for long-title-detection
- Add loading to click-to-watch
- Unstyle cursor over card

MsFairyViolet committed:
Soft Hyphen fix for Blackkklansman

rmsp reviewed (approved):
LGTM 🚀
